src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -369,6 +415,51 @@ public String getUrl() {
     return url;
   }
 
+  /**
+   * Performs all substitutions and calculates the final topic directory for a topic
+   *
+   * @param topic - String - the topic whose directory to find
+   * @return String - the directory name and path
+   */
+  public String getTopicDirFromTopic(String topic) {
+    String topicsDir = getString(TOPICS_DIR_CONFIG);
+    topicsDir = topicsDir.replace("${topic}", topic);
+
+    // only if configured
+    if (topixRegexCaptureGroup != null) {
+
+      // find all of the captured groups by the regex
+      List<String> topicGroups = new ArrayList<>();
+      Matcher matcher = topixRegexCaptureGroup.matcher(topic);
+      while (matcher.find()) {
+        if (!matcher.group().isEmpty()) {
+          topicGroups.add(matcher.group());
+        }
+      }
+
+      // make sure that all references to captured groups actually exist
+      int maxVar = extractMaximumInt(topicsDir);
```

Comment:
I don't understand this part? I thought topicsDir can contain numbers in them - "/topic_1000" by itself should be a valid topicDir? Or if the topic name contains numbers and ${topic} appears in topicDir, wouldn't your maxVar be not what you expect? I think you can do this with a regex looking for anything that looks like "\$\{\d\}" in your topicsDir and replace with what you find in topicGroups. Catch `IndexOutOfBoundsException` and rethrow the `ConfigException` for referencing non-existing capture group.

Reply:
thats actually a lot smarter
thanks!